src/adc.rs

```diff
@@ -458,6 +424,56 @@ impl_adc!(ADC1: adc_unit_t_ADC_UNIT_1);
 #[cfg(not(any(esp32c2, esp32h2, esp32c5, esp32c6, esp32p4)))] // TODO: Check for esp32c5 and esp32p4
 impl_adc!(ADC2: adc_unit_t_ADC_UNIT_2);
 
+/// Converts a raw reading to mV without using calibration
+struct DirectConverter(adc_atten_t);
+
+impl DirectConverter {
+    #[cfg(not(esp32s2))]
+    const MAX_READING: u32 = 4095;
+
+    #[cfg(esp32s2)]
+    const MAX_READING: u32 = 8191;
+
+    fn raw_to_mv(&self, raw: u16) -> u16 {
+        (raw as u32 * self.get_max_mv() as u32 / Self::MAX_READING) as u16
+    }
+
+    #[inline(always)]
+    #[allow(non_upper_case_globals)]
+    fn get_max_mv(&self) -> u16 {
+        let attenuation = self.0;
+
+        #[cfg(esp32)]
+        let mv = match attenuation {
```

Comment:
it turns out that ESP-IDF f'ed up itself. There is no ADC_ATTEN_DB_11 correctly. In truths its more a ADC_ATTEN_DB12 :D 

They removed it from the just released ESP-IDF v5.3. For more info you can look at [this](https://github.com/espressif/esp-idf/commit/0c81f41e16028db716b0f43ccbb5049e78db741b) PR

So we also need to make this work for v5.3

Reply:
They did not remove it, they renamed it. I think we need to address this separately, in a "5.3" compatibility PR or something. But I don't think that's a big deal. Seems very doable to me.